kong/core/handler.lua

```diff
@@ -119,9 +121,47 @@ return {
       end
 
       -- if set `host_header` is the original header to be preserved
-      var.upstream_host = host_header or 
+      var.upstream_host = host_header or
           balancer_address.hostname..":"..balancer_address.port
 
+      -- Keep-Alive and WebSocket Protocol Upgrade Headers
+      if var.http_upgrade == "websocket" then
```

Comment:
Sorry if you thought I implied that we should move those as well in the Lua-land. I only meant to apply changes to the PR that apply to this scope, and did not mean to add extra work to it! It would have been perfectly if this had stayed in the Nginx configuration. Maybe it still should be moved back there to avoid introducing breaking changes? As of this stands, we perform a case-sensitive check on the content of those headers, and that is a breaking change since Nginx's `map` is case-insensitive.

Sorry for the misunderstanding!